src/popsift/popsift.h

```diff
@@ -164,5 +172,8 @@ class PopSift
     int             _last_init_w; /* to support depreacted interface */
     int             _last_init_h; /* to support depreacted interface */
     ImageMode       _image_mode;
+
+    /// whether the object is initialized
+    bool            _isInit{false};
```

Comment:
Would it make sense to forbid the default constructor and copy constructor? If yes, _isInit could actually be auto-initialized to true because both constructors do that anyway.

Reply:
Yes if it does not make sense we can remove them 
```c++
    PopSift() = delete;
    PopSift(const PopSift&) = delete;
```
and set `_isInit{true}` and remove from the initialization in the cpp file.
What do u think?

Reply:
Yes, let's do that. The first constructor is actually posing as a default constructor - it has a default value for its only argument.

Reply:
done 72796c8